src/main/java/org/folio/rest/impl/TlrFeatureToggleImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.folio.rest.impl;
+
+import static io.vertx.core.Future.failedFuture;
+import static io.vertx.core.Future.succeededFuture;
+import static java.util.Collections.singletonList;
+import static org.folio.rest.jaxrs.resource.TlrFeatureToggleJobStart.PostTlrFeatureToggleJobStartResponse.respond202;
+import static org.folio.rest.jaxrs.resource.TlrFeatureToggleJobStart.PostTlrFeatureToggleJobStartResponse.respond422WithApplicationJson;
+import static org.folio.rest.jaxrs.resource.TlrFeatureToggleJobStart.PostTlrFeatureToggleJobStartResponse.respond500WithTextPlain;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.TlrFeatureToggleJobRepository;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.TlrFeatureToggleJob;
+import org.folio.rest.jaxrs.resource.TlrFeatureToggleJobStart;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.service.tlr.TlrFeatureToggleService;
+import org.folio.support.exception.TlrFeatureToggleJobAlreadyRunningException;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+public class TlrFeatureToggleImpl implements TlrFeatureToggleJobStart {
+  private static final Logger log = LogManager.getLogger(TlrFeatureToggleImpl.class);
+
+  public static final String STATUS_FIELD = "'status'";
+
+  @Override
+  public void postTlrFeatureToggleJobStart(Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    vertxContext.runOnContext(v -> handle(okapiHeaders, vertxContext)
+      .onComplete(result -> {
+        if (result.succeeded()) {
+          asyncResultHandler.handle(succeededFuture(respond202()));
```

Comment:
Maybe it makes sense always respond 202 status immediately and execute the job in a different worker pool to not block the main thread? And just log errors in case if any?

Reply:
Done